drivers/hmis_external_apis/app/views/hmis_external_apis/external_forms/form/_geolocation.haml

```diff
@@ -0,0 +1,108 @@
+:ruby
+  get_location_id = next_html_id
+  clear_id = next_html_id
+  text_id = next_html_id
+  map_id = next_html_id
+%label{for:html_id, class: required && 'required'}= label
+%br
+%button.btn.btn-secondary.mb-2{id: get_location_id}
+  Get location
+%button.btn.btn-secondary.mb-2{id: clear_id}
+  Clear
+.form-text.text-muted.mb-2{id: text_id}
+  Loading...
+%div.leaflet-container{id: map_id}
+-# A note about required behavior for Geolocation:
+-# Clicking the "Get location" button is required, but the user is allowed to clear and still submit.
+-# The user is also not blocked from submitting if geolocation errors / is unavailable.
+%input{id: html_id, hidden: true, name: name, required: required}
+.invalid-feedback This is required
+
+- content_for :page_js do
+  :javascript
+    // 'use strict';
+    $(function() {
+      var helperTextId = '#' + #{text_id.to_json.html_safe};
+      var getLocationButtonId = '#' + #{get_location_id.to_json.html_safe};
+      var clearButtonId = '#' + #{clear_id.to_json.html_safe};
+      var mapContainerId = '#' + #{map_id.to_json.html_safe};
+      var inputId = '#' + #{html_id.to_json.html_safe}
+
+      var helperText = $(helperTextId);
+      var getLocationButton = $(getLocationButtonId);
+      var clearButton = $(clearButtonId);
+      var mapContainer = $(mapContainerId);
+      var input = $(inputId);
+
+      helperText.hide();
+      mapContainer.hide();
+      clearButton.hide();
+
+      var map = L.map(#{map_id.to_json.html_safe});
```

Comment:
May want to add some more options https://leafletjs.com/reference.html#map-option but I haven't reviewed them all. I guess my first thought it that it doesn't need to be draggable, since you can't move the pin–– and it might be annoying if you are trying to scroll down the page, but you accidentally drag the map instead.
I tried `{ dragging: false, boxZoom: false, tapHold: false, touchZoom: false}`.
We can also wait on this until we have it on a device and can get some feedback!

Another thought, re: mobile ux, would be that it _might_ be nice to provide an option to hide the map once you're finished collecting geolocation. But again, we can see how it feels on device!

Reply:
Yeah that makes sense, I agree it will be better UX to disable most interaction with the map so users can scroll down the page in peace. I looked through the other options and disabled a few more. It's now still possible to zoom using the zoom in/out buttons or when focused on the element using keyboard nav. But it won't zoom or pan with normal click/tap interactions. Let's test it out on devices for sure

Reply:
perfect, thanks!